StripePaymentSheet/StripePaymentSheet/Source/Internal/FC Lite/Controllers/FCLiteAuthFlowViewController.swift

```diff
@@ -0,0 +1,234 @@
+            webView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            webView.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor),
+            webView.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor),
+            webView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+        ])
+    }
+
+    // MARK: Progress bar
+    private func setupProgressBar() {
+        let color = manifest.isInstantDebits ? FCLiteColor.link : FCLiteColor.stripe
+        progressBar.translatesAutoresizingMaskIntoConstraints = false
+        progressBar.trackTintColor = .lightGray
+        progressBar.progressTintColor = color
+        progressBar.progress = 0.0
+    }
+
+    private func showProgressBar() {
+        guard progressBar.superview == nil else { return }
+        view.addSubview(progressBar)
+        NSLayoutConstraint.activate([
+            progressBar.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            progressBar.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            progressBar.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            progressBar.heightAnchor.constraint(equalToConstant: 2.0),
+        ])
+    }
+
+    private func hideProgressBar() {
+        UIView.animate(withDuration: 0.3, delay: 0.3, options: .curveEaseOut, animations: {
+            self.progressBar.alpha = 0.0
+        }, completion: { _ in
+            self.progressBar.removeFromSuperview()
+        })
+    }
+
+    private func observeWebviewLoadingProgress() {
+        progressObservation = webView.observe(\.estimatedProgress, options: [.new]) { [weak self] _, _ in
+            guard let self else { return }
+
+            DispatchQueue.main.async { [weak self] in
+                guard let self else { return }
+                let progress = Float(self.webView.estimatedProgress)
+                self.progressBar.progress = progress
+                if progress >= 1.0 {
+                    self.hideProgressBar()
+                }
+            }
+        }
+    }
+}
+
+// MARK: WKNavigationDelegate
+extension FCLiteAuthFlowViewController: WKNavigationDelegate {
+    func webView(
+        _ webView: WKWebView,
+        decidePolicyFor navigationAction: WKNavigationAction,
+        decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
+    ) {
+        guard let url = navigationAction.request.url else {
+            return
+        }
+
+        // `matchesSchemeHostAndPath` is necessary for instant debits which
+        // contains additional query parameters at the end of the `successUrl`.
+        if url.matchesSchemeHostAndPath(of: manifest.successURL) {
+            decisionHandler(.cancel)
+            completion(.success(returnUrl: url))
+        } else if url.matchesSchemeHostAndPath(of: manifest.cancelURL) {
+            decisionHandler(.cancel)
+            completion(.cancelled)
```

Comment:
something we might want to do in the future - fetch the session even when receiving a cancel event just in case the user linked accounts and then decided to cancel